src/Scope-RedPitaya_STEMlab/main.py

```
@@ -294,66 +552,103 @@ def read_data(self):
         if isinstance(self.timerange, str):
             try:
-                self.timerange = float(self.timerange.replace("ms", "e-3").replace("µs", "e-6").replace("ns", "e-9").replace("us", "e-6").replace("s", "").replace(" ", ""))
+                self.timerange = float(
+                    self.timerange.replace("ms", "e-3")
+                    .replace("µs", "e-6")
+                    .replace("ns", "e-9")
+                    .replace("us", "e-6")
+                    .replace("s", "")
+                    .replace(" ", "")
+                )
             except:
```

Ruff (0.16.1)

[error] Do not use bare `except`

(E722)